src/Symfony/Component/Flag/BinarizedFlag.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Flag;
+
+/**
+ * Concrete Flag class that handles no-integer values.
+ *
+ * Some flags have no-integer values like this:
+ *
+ * <code>
+ * const METHOD_HEAD = 'HEAD';
+ * const METHOD_GET = 'GET';
+ * const METHOD_POST = 'POST';
+ * const METHOD_PUT = 'PUT';
+ * </code>
+ *
+ * This Flag class binarizes no-integer values internaly.
+ *
+ * @author Dany Maillard <danymaillard93b@gmail.com>
+ */
+class BinarizedFlag extends Flag
+{
+    private $map = array();
+    private $binarized = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct($from = false, $prefix = '', $bitfield = 0)
+    {
+        parent::__construct($from, $prefix, $bitfield);
+
+        if (false !== $this->from) {
+            array_walk($this->flags, function ($flag, $value) { $this->binarize($value, $flag); });
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add($flag)
```

Comment:
but then, it would be broken when using a bitmask implementation, as it would be required to support strings while it cannot